pkg/apis/pipeline/v1alpha1/taskrun_validation_test.go

```diff
@@ -0,0 +1,326 @@
+package v1alpha1
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/knative/pkg/apis"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestTaskRunValidate(t *testing.T) {
+	tests := []struct {
+		name string
+		task TaskRun
+		want *apis.FieldError
+	}{
+		{
+			name: "invalid taskspec",
+			task: TaskRun{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "taskmetaname",
+				},
+			},
+			want: apis.ErrMissingField("spec"),
+		},
+		{
+			name: "invalid taskrun metadata",
+			task: TaskRun{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "task.name",
+				},
+			},
+			want: &apis.FieldError{
+				Message: "Invalid resource name: special character . must not be present",
```

Comment:
what happens when you try to create an object like this directly via `kubectl`, does kubernetes itself catch this error?